tesla_ver/bubble_chart.py

```diff
@@ -14,185 +19,198 @@ def generateBubbleChart(server):
     app = dash.Dash(__name__, server=server, url_base_pathname="/bubblechart.html/")
     app.layout = LAYOUT
 
-    def parse_contents(contents):
-        """Parse a Dash Upload into a DataFrame.
+    @app.callback(
+        [Output("df-data", "data"), Output("df-mdata", "data")],
+        [Input("upload", "contents"), Input("upload", "filename"), Input("upload", "last_modified"),],
+    )
+    def upload_data(list_of_contents, list_of_filenames, _):
+        """This callback handles storing the dataframe as JSON in the data storage
+        component."""
 
-        contents is a string read from the upload component filename and
-        date are just information paramteres
-        """
-        _, content_string = contents.split(",")
+        df = None
 
-        decoded = base64.b64decode(content_string)
-        fileish = io.StringIO(decoded.decode("utf-8"))
-        df = pd.read_csv(fileish)
+        if None in [list_of_contents, list_of_filenames] or len(list_of_contents) != 1:
+            raise PreventUpdate
 
-        return df
+        def upload_string_to_df(content):
+            _, content_string = content.split(",")
+            decoded = base64.b64decode(content_string)
+            fileish = io.StringIO(decoded.decode("utf-8"))
+            return pd.read_csv(fileish, sep="\t")
```

Comment:
I think we should split callbacks and non-callback functions into separate files. The non-callbacks should be in a library file that is installed with when running `setup.py`

Reply:
Just to follow up from our conversation, splitting out these functions into separate files will be addressed in a future branch.